library/platform_util.c

```diff
@@ -121,24 +161,51 @@ void *mbedtls_platform_memcpy( void *dst, const void *src, size_t num )
        not normally part of the memcpy function and here can be useed
        with regular, unsecured implementation */
     memset( (void *) dst, data, num );
-    memcpy( (void *) ( (unsigned char *) dst + start_offset ),
-            (void *) ( (unsigned char *) src + start_offset ),
-            ( num - start_offset ) );
-    return( memcpy( (void *) dst, (void *) src, start_offset ) );
+
+    /* Make a copy starting from a random location. */
+    for( i = start_offset; i < num; i++ )
+    {
+        ( (char*) dst )[i] = ( (char*) src )[i];
+        flow_counter++;
+    }
+
```

Comment:
Minor: these two loops could be substituted with one using modulo.

Reply:
Replaced with one loop with modulo.